protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/DataCmdHandler.java

```diff
@@ -126,13 +137,19 @@ public void destroy() {
      *
      */
     public Response onCommand(SMTPSession session, Request request) {
-        String parameters = request.getArgument();
-        Response response = doDATAFilter(session,parameters);
-        
-        if (response == null) {
-            return doDATA(session, parameters);
-        } else {
-            return response;
+        TimeMetric timeMetric = metricFactory.timer("SMTP-" + request.getCommand());

```

Comment:
I hesitate on that point, and also on the try/finally remarks below.
I think we may enhance that latter; doing this right now will introduce a lot of try/finally which are not really needed.